criu/autofs.c

```diff
@@ -62,25 +62,57 @@ int autofs_parse(struct mount_info *pm)
 {
 	long pipe_ino = AUTOFS_OPT_UNKNOWN;
 	char **opts;
-	int nr_opts, i;
+	int nr_opts, i, ret;
 
 	split(pm->options, ',', &opts, &nr_opts);
 	if (!opts)
 		return -1;
+
 	for (i = 0; i < nr_opts; i++) {
 		if (!strncmp(opts[i], "pipe_ino=", strlen("pipe_ino=")))
-			pipe_ino = atoi(opts[i] + strlen("pipe_ino="));
+			if (xatol(opts[i] + strlen("pipe_ino="), &pipe_ino)) {
+				pr_err("pipe_ino (%s) mount option parse failed\n", opts[i] + strlen("pipe_ino="));
+				ret = -1;
+				goto free;
+			}
+	}
+
+	/*
+	 * We must inform user about bug if pipe_ino
+	 * is greater than UINT32_MAX, because, from kernel side of view
+	 * it is possible situation, because i_ino field in struct inode
```

Comment:
From kernel side of view, it is impossible situation. Despite the unsigned long  type of i_no, it can't be greater than UINT_MAX, because pipe inode numbers are generated by get_next_ino() which return type is uint.